gems/smithy-client/sig/smithy-client/configuration.rbs

```diff
@@ -0,0 +1,15 @@
+module Smithy
+  module Client
+
+    class Configuration
+      def initialize: () -> void
+      def add_option: (Symbol, ?untyped) ?{ (DefaultResolver) -> untyped } -> self
+      def build!: (?::Hash[untyped, untyped] options) -> Struct[untyped]
+
+      class DefaultResolver
```

Comment:
Ahh, so even if `DefaultResolver` is a private class - we are still defining this bit here due to the way `add_option` is defined above? Just wanted to clarify